crates/sncast/src/helpers/configuration.rs

```diff
@@ -1,7 +1,154 @@
+            keystore: get_property_optional(&entries, "keystore"),
+            wait_timeout: get_property(&entries, "wait-timeout"),
+            wait_retry_interval: get_property(&entries, "wait-retry-interval"),
+        })
+    }
+}
+
+impl Default for CastConfig {
+    fn default() -> Self {
+        Self {
+            rpc_url: String::default(),
+            account: String::default(),
+            accounts_file: Utf8PathBuf::default(),
+            keystore: None,
+            wait_timeout: WAIT_TIMEOUT,
+            wait_retry_interval: WAIT_RETRY_INTERVAL,
+        }
+    }
+}
+
+pub trait PropertyFromCastConfig: Sized {
+    fn from_toml_value(value: &Value) -> Option<Self>;
+    fn default_value() -> Self;
+}
+
+impl PropertyFromCastConfig for String {
+    fn from_toml_value(value: &Value) -> Option<Self> {
+        value.as_str().map(std::borrow::ToOwned::to_owned)
+    }
+
+    fn default_value() -> Self {
+        String::default()
+    }
+}
+
+impl PropertyFromCastConfig for Utf8PathBuf {
+    fn from_toml_value(value: &Value) -> Option<Self> {
+        value.as_str().map(Utf8PathBuf::from)
+    }
+
+    fn default_value() -> Self {
+        Utf8PathBuf::default()
+    }
+}
+
+impl PropertyFromCastConfig for u8 {
+    fn from_toml_value(value: &Value) -> Option<Self> {
+        value.as_integer().and_then(|i| i.try_into().ok())
+    }
+
+    fn default_value() -> Self {
+        WAIT_RETRY_INTERVAL
+    }
+}
+
+impl PropertyFromCastConfig for u16 {
+    fn from_toml_value(value: &Value) -> Option<Self> {
+        value.as_integer().and_then(|i| i.try_into().ok())
+    }
+
+    fn default_value() -> Self {
+        WAIT_TIMEOUT
+    }
+}
+
+impl<T> PropertyFromCastConfig for Option<T>
+where
+    T: PropertyFromCastConfig,
+{
+    fn from_toml_value(value: &Value) -> Option<Self> {
+        T::from_toml_value(value).map(Some)
+    }
+    fn default_value() -> Self {
+        Some(T::default_value())
+    }
+}
+
+pub fn get_profile(sncast_config: &Value, profile: &Option<String>) -> Result<Value> {
+    let profile_name = profile.as_deref().unwrap_or("default");
+    let config = sncast_config
+        .get("sncast")
+        .expect("Failed to find sncast config in foundry.toml file");
+
+    match config.get(profile_name) {
+        Some(profile_value) => Ok(profile_value.clone()),
+        None if profile_name == "default" => Ok(Value::Table(Default::default())),
+        None => Err(anyhow!("Profile [{}] not found in config", profile_name)),
+    }
+}
+
+pub fn get_property<T>(entries: &Value, field: &str) -> T
+where
+    T: PropertyFromCastConfig + Default,
+{
+    get_property_optional(entries, field).unwrap_or_else(T::default_value)
+}
+
+pub fn get_property_optional<T>(entries: &Value, field: &str) -> Option<T>
+where
+    T: PropertyFromCastConfig + Default,
+{
+    entries.get(field).and_then(T::from_toml_value)
+}
+
+pub fn parse_config(profile: &Option<String>, path: &Option<Utf8PathBuf>) -> Result<CastConfig> {
```

Comment:
> I don't understand what path argument is here.

This can set a path to a directory we want to start searching from. Very useful for testing, but also for taking configs from specific packages (even if we are not in the package dir)

Reply:
renamed in https://github.com/foundry-rs/starknet-foundry/pull/1644/commits/de9f39d88ada985c9bcb12ea3676aeee67b85684